controllers/datafoundationclient.go

```diff
@@ -0,0 +1,226 @@
+		)
+		isValid = false
+	}
+	if !isValid {
+		r.Log.Info("parsing ManagedFusionOffering Data Foundation Client spec failed")
+		return fmt.Errorf("invalid ManagedFusionOffering Data Foundation Client spec")
+	}
+	r.Log.Info("parsing ManagedFusionOffering Data Foundation Client spec completed successfuly")
+
+	r.dataFoundationClientSpec = dataFoundationClientSpec{
+		onboardingTicket: onboardingTicket,
+		providerEndpoint: providerEndpoint,
+	}
+	return nil
+}
+
+func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error) {
+	if !r.offering.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	} else {
+		if err := r.reconcileStorageClient(); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.reconcileCSV(); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.reconcileDefaultBlockStorageClassClaim(); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.reconcileDefaultFSStorageClassClaim(); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+}
+
+func (r *dataFoundationClientReconciler) reconcileStorageClient() error {
+	r.Log.Info("Reconciling StorageClient")
+
+	if r.dataFoundationClientSpec.providerEndpoint == "" {
+		return fmt.Errorf("dataFoundationProviderEndpoint is empty")
+	}
+
+	dfStorageProviderEndpointAsURL, err := url.ParseRequestURI(r.dataFoundationClientSpec.providerEndpoint)
+	if err != nil {
+		return err
+	}
+	if dfStorageProviderEndpointAsURL.Host == "" {
+		return fmt.Errorf("cannot retrieve host from %s, might be a malformed or partial URL", r.dataFoundationClientSpec.providerEndpoint)
+	}
+
+	if r.dataFoundationClientSpec.onboardingTicket == "" {
+		return fmt.Errorf("onboardingTicket is empty")
+
+	}
+	_, err = r.CreateOrUpdate(&r.storageClient, func() error {
+		if err := r.own(&r.storageClient, true); err != nil {
+			return err
+		}
+		r.storageClient.Spec.OnboardingTicket = r.dataFoundationClientSpec.onboardingTicket
+		r.storageClient.Spec.StorageProviderEndpoint = r.dataFoundationClientSpec.providerEndpoint
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create/update StorageClient: %v", err)
+	}
+	return nil
+}
+
+func (r *dataFoundationClientReconciler) reconcileCSV() error {
+	r.Log.Info("Reconciling OCS Client Operator CSV")
+
+	csv, err := r.getCSVByPrefix(ocsClientOperatorName)
+	if err != nil {
+		return fmt.Errorf("unable to get OCS client CSV: %v", err)
+	}
+	if err := r.own(csv, false); err != nil {
+		return err
+	}
+	if err := r.update(csv); err != nil {
+		return fmt.Errorf("failed to update OCS client CSV: %v", err)
+	}
+	return nil
+}
+
+func (r *dataFoundationClientReconciler) reconcileDefaultBlockStorageClassClaim() error {
+	r.Log.Info("Reconciling Default BlockPool StorageClassClaim")
+
+	_, err := r.CreateOrUpdate(&r.defaultBlockStorageClassClaim, func() error {
+		if err := r.own(&r.defaultBlockStorageClassClaim, true); err != nil {
+			return err
+		}
```

Comment:
you are trying to own clusterscoped resource from namespace resource i dont think it will work

Reply:
Yep, that will not work. 
We will need to be more careful in the uninstallation flow to make sure that we remove these as there is no direct ownership here.

Reply:
Removed owning the storageclassclaim